src/libdredd/src/mutation_replace_binary_operator.cc

```diff
@@ -1024,34 +1027,81 @@ MutationReplaceBinaryOperator::ClangOperatorKindToProtobufOperatorKind(
 
 bool MutationReplaceBinaryOperator::
     IsRedundantReplacementForBooleanValuedOperator(
-        clang::BinaryOperatorKind operator_kind) const {
+        clang::BinaryOperatorKind replacement_operator,
+        const clang::ASTContext& ast_context) const {
+  if (binary_operator_->getLHS()->getType()->isUnsignedIntegerType() &&
+      binary_operator_->getRHS()->getType()->isUnsignedIntegerType()) {
```

Comment:
@JonathanFoo0523 in a previous conversation I think you questioned whether it's possible for the LHS to be unsigned but not the RHS. I don't know for sure, so I'm fine with this being checked the way that it is here.